ossdbtoolsservice/query/query.py

```diff
@@ -59,8 +62,7 @@ def __init__(self, owner_uri: str, query_text: str, query_execution_settings: Qu
         self._current_batch_index = 0
         self._batches: List[Batch] = []
         self._execution_plan_options = query_execution_settings.execution_plan_options
-
-        self.is_canceled = False
+        self._cancellationToken: CancellationToken = CancellationToken()
```

Comment:
Cancellation should begin from QueryExecutionService.
Therefore, cancellation token will be generated and managed by QueryExecutionService only.

See connection service as an example.
Keep a map of query connection uri and cancellation token in QueryExecutionService.

Also,
There is no thread safety in your current code.
Use locks when you are updating the cancellationToken _canceled flag. (For reference, you can check connection service. Locking is used when updating cancellationToken maps and cancellation flags.)